src/main/java/org/spongepowered/common/mixin/core/server/management/MixinPlayerInteractionManager.java

```diff
@@ -253,27 +179,33 @@ public EnumActionResult processRightClickBlock(EntityPlayer player, net.minecraf
         }
         // Sponge End
 
-        if (!player.isSneaking() || player.getHeldItemMainhand().isEmpty() && player.getHeldItemOffhand().isEmpty()) {
+        EnumActionResult result = EnumActionResult.PASS;
+
+        boolean bypass = true;
+        final ItemStack[] itemStacks = {player.getHeldItemMainhand(), player.getHeldItemOffhand()};
```

Comment:
For a “hot” method, this array creation and for loop are unnecessary. The advantage of the old if is that it’s fail fast, this has to actually check for both and store, which is more expensive.